pkg/protocol/engine/mempool/conflictdag/conflictdag.go

```diff
@@ -28,6 +28,7 @@ type ConflictDAG[ConflictID, ResourceID IDType, VotePower VotePowerType[VotePowe
 	ConflictWeight(conflictID ConflictID) int64
 	ConflictChildren(conflictID ConflictID) (conflictIDs *advancedset.AdvancedSet[ConflictID], exists bool)
 	ConflictVoters(conflictID ConflictID) (voters map[iotago.AccountID]int64)
+	LikedInstead(conflictIDs *advancedset.AdvancedSet[ConflictID]) *advancedset.AdvancedSet[ConflictID]
```

Comment:
Here, we don't care about consistent states as we use it as a "quick filter" in the tip manager to decide if we should "rescue something" as part of the weak tip set or not (we only rescue stuff we "like").

It could of course happen that something passes this filter and we try to rescue something that we don't like but since we anyway always correct our opinion upon tip selection this doesn't really matter much - so here for performance reasons (since we need to do it for every block that is added to the tipmanager to decide where to put it, we accept these "inconsistencies" as they do no harm - we will re-evaluate our decision upon tip selection anyway and drop tips that cant be referenced efficiently.

Reply:
We currently have not a single comment anywhere in the ConflictDAG, do you really want me to add one for this method only?

I would rather go through the conflictdag and document everything at some point together (and then also document the correct use of the ReadConsistent method.